git-github-introduction.md

```diff
@@ -0,0 +1,340 @@
+
+```
+$ git clone https://github.com/john-smith/repy_v2
+Cloning into 'repy_v2'...
+remote: Counting objects: 1547, done.
+remote: Total 1547 (delta 0), reused 0 (delta 0)
+Receiving objects: 100% (1547/1547), 1.06 MiB | 0 bytes/s, done.
+Resolving deltas: 100% (778/778), done.
+Checking connectivity... done.
+```
+
+
+A **repy_v2** folder is created in the local, current working directory.  Before issuing Github commands, you need to change directories - to either the cloned repository's directory, or to one of its sub-directories. For example:
+
+```
+$ cd repy_v2/
+```
+
+### Set Alias for Forked Repository
+Setting an alias for a repository can help developers better manage repository changes and
+keep local and remote repositories in sync.  The alias is a string that refers to the repository by name instead of the
+repository's full URL.  In the example below, the forked [repy_v2](https://github.com/SeattleTestbed/repy_v2) repository is given the **upstream** alias:
+
+
+```
+$ git remote add upstream https://github.com/SeattleTestbed/repy_v2.git
+```
+
+
+### Retrieve Latest Version of Forked Repository
+
+Fetching and merging the latest changes of `upstream` (remembering that the `upstream` alias points to the forked repository) can be done with the
+`pull` Github command.  The `pull` command fetches changes made to the remote repository and merges them into your local repository.  Branches are multiple working versions of a master project (or document). The main branch is the master repository and multiple, local branches of this master may be created by users so that they can work on the project before committing their work to the master document.  The pull command expects to see a (by default, the master) branch name, which the user will find on the project's [main repository](https://github.com/SeattleTestbed/repy_v2).  Information on creating a new branch is in the next section, ''Add a New Branch Locally''.
+
+![GitHub Branches](/images/branches_sm.png)
+
+
+Additional information on branching: [branching](http://git-scm.com/book/en/Git-Branching-What-a-Branch-Is)
+
+```
+$ git pull upstream master
+From https://github.com/SeattleTestbed/repy_v2
+ * branch            master     -> FETCH_HEAD
+  * [new branch]      master     -> upstream/master
+  Already up-to-date.
+```
+
+### Add a New Branch Locally
+
+First, list all known local branches of the repository, by typing the following:  
+
+```
+$ git branch
+* master
+```
+
+Next, to add a new branch, which is the preferred method for working on and adding new features, type **git branch new_branch**, where **''new_branch**'' is the name of the new branch:
+
+```
+$ git branch new_branch
+```
+
+The next step is to switch to the newly created branch:
+
+```
+$ git checkout new_branch
+Switched to branch 'new_branch'
+```
+
+

```

Comment:
Probably it's worth mentioning that `git checkout -b [branchname]` can do both steps at once...